### PR TITLE
[4.1.0] Modify exception handling to continue migration on errors

### PR DESCRIPTION
**Pull request by Lakith-Rambukkanage**
## Purpose
When migrations fail due to errors, only the first occurrence is identified due to exiting tenant/artefact wise iterations. Exception handling is needed to be modified to continue on errors to identify all errors at once.
Fixing : https://github.com/wso2/api-manager/issues/615
Fixing : https://github.com/wso2/api-manager/issues/801
Fixing : https://github.com/wso2/api-manager/issues/802

## Goals
Identify all errors in one migration attempt.

## Approach
Exceptions thrown within an iteration is caught and logged without exiting the loop. 
Error state is maintained and logged at the end.

## Related PRs
#254 
#300

### CODE REVIEW

File: components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/v300/V300RegistryResourceMigrator.java [hunk 1]
```diff
@@ -54,14 +54,16 @@ public void migrate() throws APIMigrationException {
...
         for (Tenant tenant : tenants) {
-            try {
-                registryService.startTenantFlow(tenant);
-                log.info("WSO2 API-M Migration Task : Updating APIs of tenant " + tenant.getId() +
-                        '(' + tenant.getDomain() + ')');
-                GenericArtifact[] artifacts = registryService.getGenericAPIArtifacts();
-                for (GenericArtifact artifact : artifacts) {
+            registryService.startTenantFlow(tenant);
+            log.info("WSO2 API-M Migration Task : Updating APIs of tenant " + tenant.getId() +
+                    '(' + tenant.getDomain() + ')');
```
Comment (dushaniw) [inline components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/v300/V300RegistryResourceMigrator.java:63]:
```suggestion
                   +  '(' + tenant.getDomain() + ')');
```

File: components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/v300/V300RegistryResourceMigrator.java [hunk 2]
```diff
@@ -54,14 +54,16 @@ public void migrate() throws APIMigrationException {
...
+        log.info("WSO2 API-M Migration Task : Started Updating API artifacts for all tenants");
         for (Tenant tenant : tenants) {
-            try {
-                registryService.startTenantFlow(tenant);
-                log.info("WSO2 API-M Migration Task : Updating APIs of tenant " + tenant.getId() +
-                        '(' + tenant.getDomain() + ')');
-                GenericArtifact[] artifacts = registryService.getGenericAPIArtifacts();
-                for (GenericArtifact artifact : artifacts) {
+            registryService.startTenantFlow(tenant);
+            log.info("WSO2 API-M Migration Task : Updating APIs of tenant " + tenant.getId() +
```
Comment (dushaniw) [inline components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/v300/V300RegistryResourceMigrator.java:62]: in wso2 code style, + should comes to the next line if statement goes to more than one line.
```suggestion
            log.info("WSO2 API-M Migration Task : Updating APIs of tenant " + tenant.getId()
```

File: components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/v320/V320DBDataMigrator.java [hunk 3]
```diff
@@ -66,11 +67,14 @@ public void migrate() throws APIMigrationException {
...
      */
     private void updateAPITypeInDb() throws APIMigrationException {
+        boolean isError = false;
+        log.info("WSO2 API-M Migration Task : Started updating API Type in DB for all tenants");
         tenantManager = ServiceHolder.getRealmService().getTenantManager();
-
         try {
             List<Tenant> tenants = APIUtil.getAllTenantsWithSuperTenant();
             for (Tenant tenant : tenants) {
+                log.info("WSO2 API-M Migration Task : Started updating API Type in DB " +
```
Comment (dushaniw) [inline components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/v320/V320DBDataMigrator.java:76]: according to code style followed by WSO2 '+' sign should precede the line.

Comment (dushaniw) [reply components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/v320/V320DBDataMigrator.java:76]: lets fix this in all the places.

Comment (Lakith-Rambukkanage) [inline components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/v320/V320DBDataMigrator.java:76]:
```suggestion
                log.info("WSO2 API-M Migration Task : Started updating API Type in DB "
```

File: components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/v320/V320DBDataMigrator.java [hunk 4]
```diff
@@ -66,11 +67,14 @@ public void migrate() throws APIMigrationException {
...
     private void updateAPITypeInDb() throws APIMigrationException {
+        boolean isError = false;
+        log.info("WSO2 API-M Migration Task : Started updating API Type in DB for all tenants");
         tenantManager = ServiceHolder.getRealmService().getTenantManager();
-
         try {
             List<Tenant> tenants = APIUtil.getAllTenantsWithSuperTenant();
             for (Tenant tenant : tenants) {
+                log.info("WSO2 API-M Migration Task : Started updating API Type in DB "
+                        "for tenant: " + tenant.getDomain());
```
Comment (Lakith-Rambukkanage) [inline components/migration-client/wso2-api-migration-client/src/main/java/org/wso2/carbon/apimgt/migration/migrator/v320/V320DBDataMigrator.java:77]:
```suggestion
                        + "for tenant: " + tenant.getDomain());
```

**Review by dushaniw — CHANGES_REQUESTED** [review]

**Review by dushaniw — APPROVED** [review]

**Pull request merged**